src/models/time-sync.ts

```diff
@@ -136,15 +138,70 @@ export class TimeSyncModel implements ModelPluginInterface {
     }
   }
 
+  private checkPadding(inputs: ModelParams[]): boolean[] {
+    const out = [false, false];
+    if (this.startTime !== undefined) {
+      const startDiffInSeconds =
+        moment(inputs[0].timestamp).diff(moment(this.startTime)) / 1000;
+      const endDiffInSeconds =
+        moment(inputs[inputs.length - 1].timestamp)
+          .add(inputs[inputs.length - 1].duration, 'seconds')
+          .diff(moment(this.endTime)) / 1000;
+      if (startDiffInSeconds > 0) {
+        out[0] = true;
+      }
+      if (endDiffInSeconds < 0) {
+        out[1] = true;
+      }
+    }
+    return out;
+  }
+
+  private async padInputs(
+    inputs: ModelParams[],
+    pad: boolean[]
+  ): Promise<ModelParams[]> {
+    if (pad[0]) {
+      const dateRange = momentRange.range(
+        moment(this.startTime),
+        moment(inputs[0].timestamp).subtract(1, 'second')
+      );
+      for (const second of dateRange.by('second')) {
+        // @todo apply zero-fill logic to create this object (any fields with aggregation-method == sum or avg should be zeros, others copied from inputs[0])
+        inputs.push({timestamp: second.toISOString(), duration: 1, carbon: 0});
+      }
+      inputs.sort((a, b) => moment(a.timestamp).diff(moment(b.timestamp)));
+    }
+    if (pad[1]) {
+      const dateRange = momentRange.range(
+        moment(inputs[inputs.length - 1].timestamp).add(
+          inputs[inputs.length - 1].duration + 1,
+          'seconds'
+        ),
+        moment(this.endTime)
+      );
+      console.log('DATE RANGE ', dateRange);
+      console.log(inputs);
+      for (const second of dateRange.by('second')) {
+        // @todo apply zero-fill logic to create this object (any fields with aggregation-method == sum or avg should be zeros, others copied from inputs[0])
+        inputs.push({timestamp: second.toISOString(), duration: 1, carbon: 0});
+      }
+      inputs.sort((a, b) => moment(a.timestamp).diff(moment(b.timestamp)));
```

Comment:
sorting is not applied to inputs since returned value is not used

Reply:
oh good point - fixing now